src/arfs/arfs_builders/arfs_builders.ts

```diff
@@ -117,13 +123,63 @@ export abstract class ArFSMetadataEntityBuilder<T extends ArFSEntity> {
 	}
 
 	async build(node?: GQLNodeInterface): Promise<T> {
-		await this.parseFromArweaveNode(node, this.owner);
+		const extraTags = await this.parseFromArweaveNode(node, this.owner);
+		this.parseCustomMetaDataFromGqlTags(extraTags);
+
 		return this.buildEntity();
 	}
 
+	private parseCustomMetaDataFromGqlTags(gqlTags: GQLTagInterface[]): void {
+		for (const { name, value } of gqlTags) {
+			this.addToCustomMetaData({ [name]: value });
+		}
+	}
+
+	protected addToCustomMetaData(customMetaData: EntityMetaDataTransactionData): void {
+		if (!isCustomMetaDataJsonFields(customMetaData)) {
+			console.error(`Parsed an invalid custom metadata shape from MetaData Tx Data JSON: ${customMetaData}`);
+			return;
+		}
+
+		for (const key of Object.keys(customMetaData)) {
+			let customMetaDataValue: JsonSerializable | JsonSerializable[];
+
+			const prevValue = this.customMetaData[key];
+			const newValue = customMetaData[key];
+
+			if (prevValue) {
+				customMetaDataValue = Array.isArray(prevValue) ? prevValue : [prevValue];
+
+				if (Array.isArray(newValue)) {
+					for (const val of newValue) {
+						customMetaDataValue.push(val);
+					}
+				} else {
+					customMetaDataValue.push(newValue);
+				}
+			} else {
+				customMetaDataValue = newValue;
+			}
+
+			this.customMetaData[key] = customMetaDataValue;
+		}
```

Comment:
We could remove lots of indentation by normalizing the values as arrays at the beginning 
```suggestion
		for (const key of Object.keys(customMetaData)) {
			const prevValue = this.customMetaData[key] || [];
			const newValue = customMetaData[key];

			const customMetaDataValue: JsonSerializable | JsonSerializable[] = Array.isArray(prevValue)
				? prevValue
				: [prevValue];
			const newValueAsArray = Array.isArray(newValue) ? newValue : [newValue];

			for (const val of newValueAsArray) {
				customMetaDataValue.push(val);
			}

			this.customMetaData[key] = customMetaDataValue;
		}
```

Reply:
So:
1. `prevValue` is a non-nullable array (that gets rid of the `if (prevValue)`)
2. we declare `newValueAsArray` so we'll have to simply iterate the array with no `if`s

Then:
- when there is no items in `prevValue`, it simply is an empty array;
  - regardless of if it contains items or not, the newValues will be merged into the array
- regardless of `newValue` being a `string` or `string[]`, it will simply loop `newValueAsArray`

Reply:
We can even subtract this piece of code into its own getter `get customMetadata`

Reply:
I really like this suggestion! Thanks @matibat

Reply:
This doesn't handle every permutation

Reply:
If newValue is not an array and prevValue is undefined, we expect this.customMetaData[key] to become newValue not [newValue]

Reply:
This is a waste of time

Reply:
It seemed equivalent to me. But you are right, there are cases working unexpectedly